src/client/entity-editor/common/entity.js

```diff
@@ -34,31 +35,30 @@ type EntityProps = {
 function Entity(
 	{disambiguation, id, text, type, unnamedText}: EntityProps
 ) {
+	function eventHandlerk(event, url) {
+		event.preventDefault();
+		window.open(url, '_blank');
+	}
 	const url = type === 'Area' ?
 		`//musicbrainz.org/area/${id}` :
 		`/${type.toLowerCase()}/${id}`;
-	// const nameComponent = text || <i>{unnamedText}</i>;
+	const nameComponent = text || <i>{unnamedText}</i>;
 	const contents = (
 		<span>
 			{
 				type && genEntityIconHTMLElement(type)
 			}
 			{' '}
-			<a href={url} rel="noopener noreferrer" target="_blank">
-				{text} &nbsp; ( {type} )
-			</a>
-
+			<section onClick={eventHandlerk(event, url)}>
```

Comment:
Here, in your onClick property, you actually call the function eventHandlerk instead of passing the function itself (`onClick={function(arguments)}` instead of `onClick={eventHandlerk}`).

That's why a new tab was opening: the function was called everytime the component was rendered!
That's an important distinction to keep in mind. I know I made that mistake before!

You don't need to pass arguments. The onClick function will call your callback with the first argument `event`, and the `url` is defined in the render function, no need to pass it to `eventHandlerk`.

Reply:
A few other comments:

- You previously had an `<a>` tag instead of the `<section>`. It was better with an `<a>`: specific styling is applied that lets the user know it's a clickable link.
- What does the `k` stand for at the end of `eventHandlerk`?
- I touched on the subject before, but you'll want to do all these modifications on a new component (I suggest the name `SearchResultEntity`) because this one is used in other places and we don't want these modifications everywhere.